### PR TITLE
[google_maps_flutter_web] Add padding to newLatLngBounds implementation

### PR DESCRIPTION
**Pull request by jxstxn1**
## Description
This PR is implementing the `padding` argument to `fitBounds` which is called by `newLatLngBounds`.

## Related Issues
[flutter/flutter#122192](https://github.com/flutter/flutter/issues/122192)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
@ditman Can you do the initial review here? It looks like this one should be pretty quick.

**Comment by jxstxn1** [comment]
@stuartmorgan @ditman Is there any update here?

**Comment by stuartmorgan** [comment]
Update from triage: This is just waiting for @ditman to have bandwidth to review maps PRs.

File: packages/google_maps_flutter/google_maps_flutter_web/example/integration_test/projection_test.dart
Comment (ditman) [inline packages/google_maps_flutter/google_maps_flutter_web/example/integration_test/projection_test.dart:126]: Can we have a test where padding != 0, and verify that the southWest and northEast are actually NOT in the borders of the map, but actually closer to the center, because of the value of the `padding`?

**Review by ditman — COMMENTED** [review]
Thanks for cleaning this up! Apart of a version update to clear the conflicts, if possible it'd be nice to have a test that actually verifies that the `padding` value is doing something with the rendering of the map. Other than that minor nitpick, this should be ready to go!

File: packages/google_maps_flutter/google_maps_flutter_web/example/integration_test/projection_test.dart
Comment (jxstxn1) [reply packages/google_maps_flutter/google_maps_flutter_web/example/integration_test/projection_test.dart]: Implemented it and also reduced the test size
Hope this is fine this way :)

File: packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md
Comment (ditman) [inline packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md:3]:
```suggestion
* Adds padding support to `CameraUpdate.newLatLngBounds`. Issue [#122192](https://github.com/flutter/flutter/issues/122192).
```

**Review by ditman — COMMENTED** [review]
Thanks for the new test, I like it better than before and actually exercises the new code paths. `LatLngBounds` has an overridden `==` operator, so the comparisons should work as intended.

Very minor change, but please format the changelog using the [style guide](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style). (I've suggested a new line, but feel free to rewrite however sounds best to you)

**Review by ditman — APPROVED** [review]
LGTM!

Please @stuartmorgan, do take a look for a second review here (you've dealt with `convert.dart` before :P)

**Review by stuartmorgan — APPROVED** [review]

**Pull request merged**
